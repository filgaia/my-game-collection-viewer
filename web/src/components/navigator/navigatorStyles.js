const navigatorStyles = (theme) => ({
    appBar: {
        position: 'relative',
    },
    grow: {
        flexGrow: 1,
    },
    icon: {
        marginRight: theme.spacing.unit * 2,
    }
});

export default navigatorStyles;
