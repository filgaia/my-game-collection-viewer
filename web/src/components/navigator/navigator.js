// @vendors
import React, { Component } from 'react';
import PropTypes from 'prop-types';
// @material
import {
  AppBar,
  IconButton,
  Toolbar,
  Tooltip,
  Typography,
} from '@mui/material';
// https://material.io/tools/icons/?style=baseline
import {
  Games,
  SortByAlpha,
} from '@mui/icons-material';
import withStyles from '@mui/styles/withStyles';
// @components
import ImportDrawer from '../importDrawer/importDrawer';
import FilterDrawer from '../filterDrawer/filterDrawer';
// @styles
import navigatorStyles from './navigatorStyles';

export class Navigator extends Component {
  render() {
    const {
      classes, importFile, gamesInformation, setLabelFilter, shortByName, toggleDrawer,
    } = this.props;

    return (
      <AppBar position="static" className={classes.appBar}>
        <Toolbar>
          <Games className={classes.icon} />
          <Typography variant="h6" color="inherit" noWrap className={classes.grow}>
            My Game Collection Viewer
          </Typography>
          <IconButton color="inherit" onClick={shortByName} size="large">
            <Tooltip title="Order list" aria-label="Order list">
              <SortByAlpha />
            </Tooltip>
          </IconButton>
          <FilterDrawer
            filterDrawer={gamesInformation.get('filterDrawer')}
            gamesInformation={gamesInformation}
            setLabelFilter={setLabelFilter}
            toggleDrawer={toggleDrawer}
          />
          <ImportDrawer
            importDrawer={gamesInformation.get('importDrawer')}
            importFile={importFile}
            toggleDrawer={toggleDrawer}
          />
        </Toolbar>
      </AppBar>
    );
  }
}

Navigator.propTypes = {
  classes: PropTypes.object.isRequired,
  gamesInformation: PropTypes.object.isRequired,
  importFile: PropTypes.func.isRequired,
  setLabelFilter: PropTypes.func.isRequired,
  shortByName: PropTypes.func.isRequired,
  toggleDrawer: PropTypes.func.isRequired,
};

export default withStyles(navigatorStyles)(Navigator);
