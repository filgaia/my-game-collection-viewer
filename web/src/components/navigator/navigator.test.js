import React from 'react';
import renderer from 'react-test-renderer';
import { Navigator } from './navigator';

describe('<Navigator />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Navigator
        classes={{}}
        gamesInformation={{ get: jest.fn() }}
        importFile={jest.fn()}
        setLabelFilter={jest.fn()}
        shortByName={jest.fn()}
        toggleDrawer={jest.fn()}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
