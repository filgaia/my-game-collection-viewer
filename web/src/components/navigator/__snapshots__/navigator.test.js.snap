// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigator /> renders correctly 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="GamesIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M15 7.5V2H9v5.5l3 3 3-3zM7.5 9H2v6h5.5l3-3-3-3zM9 16.5V22h6v-5.5l-3-3-3 3zM16.5 9l-3 3 3 3H22V9h-5.5z"
      />
    </svg>
    <h6
      className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-2l1qg7-MuiTypography-root"
    >
      My Game Collection Viewer
    </h6>
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        aria-label="Order list"
        aria-labelledby={null}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-mui-internal-clone-element={true}
        data-testid="SortByAlphaIcon"
        focusable="false"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M14.94 4.66h-4.72l2.36-2.36zm-4.69 14.71h4.66l-2.33 2.33zM6.1 6.27 1.6 17.73h1.84l.92-2.45h5.11l.92 2.45h1.84L7.74 6.27H6.1zm-1.13 7.37 1.94-5.18 1.94 5.18H4.97zm10.76 2.5h6.12v1.59h-8.53v-1.29l5.92-8.56h-5.88v-1.6h8.3v1.26l-5.93 8.6z"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        aria-label="Filter..."
        aria-labelledby={null}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-mui-internal-clone-element={true}
        data-testid="FilterListIcon"
        focusable="false"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        aria-label="Add Games..."
        aria-labelledby={null}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-mui-internal-clone-element={true}
        data-testid="CloudDownloadIcon"
        focusable="false"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM17 13l-5 5-5-5h3V9h4v4h3z"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</header>
`;
